### PR TITLE
CNDIT-1103 : Spin out the Organization Service as a separate MicroService

### PR DESCRIPTION
**Pull request by alog-enquizit**
- Separate out the Organization processing logic to the Organization MicroService 
- Add all the scaffolding i.e., data model skeleton, Kafka Stream Processing logic, Controller, Service, Basic Unit Tests

### CODE REVIEW

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/patient/PatientSp.java [hunk 1]
```diff
@@ -7,19 +7,23 @@
...
 
 @Slf4j
 @Data
+@Builder
 @Entity
 @NoArgsConstructor
+@AllArgsConstructor
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class Patient {
+public class PatientSp {
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/patient/PatientSp.java:23]: Could we add a comment in a later commit or so at the start of the file to add some context to what this dto is for?

Comment (upasanapattnaik-eq) [reply data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/patient/PatientSp.java:23]: That being said, I like the separation of output objects

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Address.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private String cntyCd;
+    private String state;
+    private String cntryCd;
+    private String state_desc;
+    private String county;
+    private String withinCityLimitsInd;
+    private String country;
+    private String addressComments;
+
+    public <T> T updateOrg(T org) {
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Address.java:34]: Does the addr_elp_cd, addr_elp_use_cd and addr_pl_uid get pulled in via this section?
There is a check for addr_elp_use_cd = 'WP', addr_elp_cd = 'O'. ELP class_cd ='PST' is handled in the sproc.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Fax.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@ToString(callSuper = true)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class Fax implements OrgExtension {
+    private Long faxTlUid;
+    private String faxElpCd;
+    private String faxElpUseCd;
+    private String orgFax;
+
+    public <T> T updateOrg(T org) {
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Fax.java:23]: Just for context, fax specific codes have been implemented in the sproc as they the section is fax specic but for reference- 
fax_elp_cd = 'FAX'
fax_elp_use_cd = 'WP'
These codes should be coming through. 
ELP class_cd = 'TELE' is not part of the json.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Phone.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+public class Phone implements OrgExtension {
+    private Long phTlUid;
+    private String phElpCd;
+    private String phElpUseCd;
+    private String telephoneNbr;
+    private String extensionTxt;
+    private String emailAddress;
+    private String phone_comments;
+
+    public <T> T updateOrg(T org) {
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Phone.java:26]: I'll leave the ELP codes for ref
ph_elp_cd = 'PH'
ph_elp_use_cd = 'WP'

elp class_cd = 'TELE' is not sent in the json.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    private String assigningAuthorityCd;
+
+    public <T> T updateOrg(T org) {
+        if (org.getClass() == OrgReporting.class) {
+            OrgReporting orgReporting = (OrgReporting) org;
+            orgReporting.setQuickCode(rootExtensionTxt);
+            //ToDo: Revisit logic
+            orgReporting.setFacilityId(rootExtensionTxt);
+            //ToDo: Revisit logic
+            orgReporting.setFacilityIdAuth(rootExtensionTxt);
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java:34]: I believe this should be the ASSIGNING_AUTHORITY_CD once the code filters pass through for FacilityId

Comment (upasanapattnaik-eq) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java:34]: There is also an SRTE join with this information. I will update that additional logical piece in the sproc. Null if none of the conditions are met, else value for FACILITY_ID_AUTH.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    private String recordStatusCd;
+    private String rootExtensionTxt;
+    private String entityIdSeq;
+    private String assigningAuthorityCd;
+
+    public <T> T updateOrg(T org) {
+        if (org.getClass() == OrgReporting.class) {
+            OrgReporting orgReporting = (OrgReporting) org;
+            orgReporting.setQuickCode(rootExtensionTxt);
+            //ToDo: Revisit logic
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java:31]: Quick code TYPE_CD = 'QEC'

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/patient/PatientSp.java (hunk 1 above)
Comment (alog-enquizit) [reply data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/patient/PatientSp.java:23]: Sure thing. Will do.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            orgReporting.setFacilityId(rootExtensionTxt);
+            //ToDo: Revisit logic
```
[the remark this answers is not in the document]
Comment (alog-enquizit) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java]: Updated.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java (hunk 6 above)
Comment (alog-enquizit) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java]: Updated.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java (hunk 5 above)
Comment (alog-enquizit) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Entity.java]: Updated.

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Phone.java (hunk 4 above)
Comment (upasanapattnaik-eq) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Phone.java:26]: Updated sproc

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Fax.java (hunk 3 above)
Comment (upasanapattnaik-eq) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Fax.java:23]: Updated sproc

File: data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Address.java (hunk 2 above)
Comment (upasanapattnaik-eq) [reply data-reporting-service/organization-service/src/main/java/gov/cdc/etldatapipeline/organization/model/dto/orgdetails/Address.java:34]: Updated sproc.

**Review by rshanmugam-enquizit — CHANGES_REQUESTED** [review]
Can you split this in to two PRs? I see both organization as well as changes to person service and changes to 89 files needs to be atleast 3-4 PRs if able

**Comment by alog-enquizit** [comment]
> Can you split this in to two PRs? I see both organization as well as changes to person service and changes to 89 files needs to be atleast 3-4 PRs if able

@rshanmugam-enquizit The major changes in the Person service is with respect to the Organization service migration into its own microservice. The other changes in Person Service are implementing the builder pattern and the renaming of the files. Nevertheless split the changes to two separate PRs:
https://github.com/CDCgov/NEDSS-DataAccess/pull/57 (Person Service)
https://github.com/CDCgov/NEDSS-DataAccess/pull/58 (Organization Service)

**Comment by alog-enquizit** [comment]
This PR was forked into 
https://github.com/CDCgov/NEDSS-DataAccess/pull/57
https://github.com/CDCgov/NEDSS-DataAccess/pull/58

Closing this PR.

**Pull request closed without merging**